okhttp/src/main/java/okhttp3/internal/Platform.java

```diff
@@ -274,6 +298,55 @@ public Android(Class<?> sslParametersClass, OptionalMethod<Socket> setUseSession
   }
 
   /**
+   * OpenJDK 9+.
+   */
+  private static class Jdk9Platform extends Platform {
+    private final Method setProtocolMethod;
+    private final Method getProtocolMethod;
+
+    public Jdk9Platform() {
+      try {
+        this.setProtocolMethod =
+            SSLParameters.class.getMethod("setApplicationProtocols", String[].class);
+        this.getProtocolMethod = SSLSocket.class.getMethod("getApplicationProtocol");
+      } catch (NoSuchMethodException e) {
+        throw new AssertionError();
+      }
+    }
+
+    @Override public void configureTlsExtensions(SSLSocket sslSocket, String hostname,
+        List<Protocol> protocols) {
+      try {
+        SSLParameters p = sslSocket.getSSLParameters();
+
+        List<String> names = alpnProtocolNames(protocols);
+
+        setProtocolMethod.invoke(p,
+            new Object[] {names.toArray(new String[names.size()])});
+
+        sslSocket.setSSLParameters(p);
+      } catch (IllegalAccessException | InvocationTargetException e) {
+        throw new AssertionError();
+      }
+    }
+
+    @Override public String getSelectedProtocol(SSLSocket socket) {
+      try {
+        String protocol = (String) getProtocolMethod.invoke(socket);
+
+        if (protocol != null && protocol.equals("")) {
```

Comment:
I think this line might want to be `if (protocol == null || protocol.equals(""))` . . . either of these cases is symmetric